magento/pwa-studio
# [PWA-2306]: Remove deprecated "id" field from customer queries

## Description

PR to remove all `customer.id` fields in graphQL queries as it is deprecated. This also adds a merge function to Customer typePolicy so it will always merge incoming objects on Apollo cache, this change was necessary due to the lack of a unique identifier on `Customer` type. 

## Related Issue







Closes PWA-2306

## Acceptance





1. PWA Studio no longer requests the `id` field on `Customer` objects
2. `Customer` objects are still cached appropriately
3. There is no performance slowdown on authenticated routes like wishlist, saved payments, etc.


### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature





1. Verify that creating an account still functions correctly (regression test)
2. Verify that signing in still functions correctly (regression test)
3. Verify that the Communications Page still functions correctly (regression test)
4. Verify that the Order History Page still functions correctly (regression test)
5. Verify that the Wishlist Page still functions correctly (regression test)
6. Verify that the Account Information Page still functions correctly (regression test)
7. Verify that the Checkout Page still functions correctly (regression test)
8. Verify that Address Book functions correctly
9. Verify that the Shipping Information component pulls accurate details from the Customer to pre-populate its fields
10. Verify that navigating around the site as an authenticated user functions correctly (regression test)

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
